tree_test.go

```diff
@@ -1649,3 +1654,127 @@ func TestLeafNodeInsert(t *testing.T) {
 		t.Fatalf("hash should not be nil")
 	}
 }
+
+type randTest []randTestStep
+
+type randTestStep struct {
+	op    int
+	key   []byte // for opInsert, opDelete, opGet
+	value []byte // for opInsert
+	err   error  // for debugging
+}
+
+const (
+	opInsert = iota
+	opDelete
+	opGet
+	opHash
+	opCommit
+	opProve
+	opMax
+)
+
+func (randTest) Generate(r *mRand.Rand, size int) reflect.Value {
+	var finishedFn = func() bool {
+		size--
+		return size == 0
+	}
+	return reflect.ValueOf(generateSteps(finishedFn, r))
+}
+
+func generateSteps(finished func() bool, r io.Reader) randTest {
+	var allKeys [][]byte
+	var one = make([]byte, 32)
+	genKey := func() []byte {
+		r.Read(one)
+		if len(allKeys) < 2 || one[0]%100 > 90 {
```

Comment:
Maybe add a comment to help the reader, like: ~"If these are the first two operations, we force inserting keys. If that isn't the case, make add a new key 10% of the time or return an existing key otherwise".